assets/model_monitoring/components/src/action_analyzer/action_detector_component/run.py

```diff
@@ -0,0 +1,166 @@
+    """Get the violated metrics names from the gsq output.
+
+    Args:
+        signal_out_url(str): gsq output url.
+        signal_name(str): signal name defined by user.
+    Returns:
+        List[str]: list of violated metrics.
+    """
+    violated_metrics = []
+    try:
+        store_url = StoreUrl(signal_out_url)
+        gsq_output = store_url.read_file_content(f"{signal_name}.json")
+        gsq_output_json = json.loads(gsq_output)
+        metrics_dict = gsq_output_json["metrics"]
+        for metrics in GSQ_METRICS_LIST:
+            pass_rate_metrics = f"Aggregated{metrics}PassRate"
+            if pass_rate_metrics in metrics_dict:
+                if metrics_dict[pass_rate_metrics]["value"] < metrics_dict[pass_rate_metrics]["threshold"]:
+                    print(f"Metrics {metrics} violated.")
+                    violated_metrics.append(metrics)
+        return violated_metrics
+    except Exception as e:
+        print("Exception while getting the violated metrics. ", e)
+        return []
+
+
+def run():
+    """Script for action detector component."""
+    # Parse argument
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--action_output", type=str)
+    parser.add_argument("--signal_scored_data", type=str)
+    parser.add_argument("--signal_output", type=str)
+    parser.add_argument("--signal_name", type=str)
+    parser.add_argument("--model_deployment_name", type=str, required=True)
+    parser.add_argument("--workspace_connection_arm_id", type=str, required=True)
+    parser.add_argument("--aml_deployment_id", type=str)
+    parser.add_argument("--query_intention_enabled", type=str)
+    args = parser.parse_args()
+
+    # get violated metrics
+    violated_metrics = get_violated_metrics(args.signal_output, f"signals/{args.signal_name}")
+    if violated_metrics == []:
+        print("No violated metrics. No action will be generated.")
+        return
+    print("Violated metrics found: ", violated_metrics)
+
+    signal_scored_data_df = try_read_mltable_in_spark(args.signal_scored_data, "signal_scored_data")
+    print("gsq output df")
+    signal_scored_data_df.show()
+    df_pandas = signal_scored_data_df.toPandas()
+
+    # find unique indexes
+    unique_indexes = get_unique_indexes(df_pandas)
+    if not unique_indexes or len(unique_indexes) == 0:
+        print("No index found in the data input. No action will be generated.")
+        return
+    print(f"Found {len(unique_indexes)} indexes in the input data.")
+
+    llm_client = LLMClient(args.workspace_connection_arm_id, args.model_deployment_name)
+
+    # detect actions for each index
+    final_actions = []
+    for index in unique_indexes:
+        index_actions = []
+        # low retrieval score index action detector
+        lrsi_action_detector = LowRetrievalScoreIndexActionDetector(index,
+                                                                    violated_metrics,
```

Comment:
These code should be generic, calling from the interface

Reply:
what do you mean?